src/wdio/services/TerraCommands/determine-test-options.js

```diff
@@ -0,0 +1,82 @@
+/**
+* Helper method to determine the default test options for one or two args passed.
+* @param {Array} args - The list of test arguments to parse.
+*     Supports [ String ], [ object ], [ string, object ]
+*/
+const determineArgs = (args) => {
+  const param1 = args.length ? args[0] : undefined;
+  const param2 = args.length > 1 ? args[1] : undefined;
+
+  let name = 'default';
+  let options = {};
+  if (typeof param1 === 'string') {
+    name = param1;
+    options = typeof param2 === 'object' && !Array.isArray(param2) ? param2 : options;
+  } else {
+    options = typeof param1 === 'object' && !Array.isArray(param1) ? param1 : options;
+  }
+
+  // Check if custom selector should be used, otherwise use the global value.
+  const selector = options.selector || global.browser.options.terra.selector;
+
+  // Checked for custom axe selector
+  const { context } = options;
+
+  // Which viewports the screenshoot should adjust too during the test run, otherwise use's current viewport
+  const { viewports } = options;
+
+  return {
+    context,
+    name,
+    options,
+    selector,
+    viewports,
+  };
+};
+
+/**
+* Helper method to determine the selector and axe rules for the test run.
+* @param {Array} args - The list of test arguments to parse.
+*/
+const axeOptions = (args) => {
+  const {
+    context, options, selector, viewports,
+  } = determineArgs(args);
+
+  const axeRules = options.rules || options.axeRules;
+
+  return {
+    context: context || selector,
```

Comment:
should selector take precedence?

Reply:
Currently the `Terra.should.beAccessible()` only accepts a context option for identifying the element to be tested, however the `Terra.should.valdiateScreenshot()` only accepts a selector. This essentially will use context if defined, otherwise it uses the selector. This could leave a hole where someone could pass both context and selector to Terra.should.validateElement.... maybe should pass both back? 

@mjhenkes @ryanthemanuel @yuderekyu should we attempt to handle both? Should we rename the beAccessible element identifier to `selector` from `context`? This seems like a pain but solved with `ctrl + f`. I can update the beAccessible to fallback to selector is context is invalid?

Reply:
i'm cool with renaming `context` to `selector` since we abstract axe away